dace/transformation/passes/scalar_to_symbol.py

```diff
@@ -322,6 +322,7 @@ def __init__(self, in_edges: Dict[str, mm.Memlet], out_edges: Dict[str, mm.Memle
 
     def visit_Subscript(self, node: ast.Subscript) -> Any:
         # Convert subscript to symbol name
+        node = self.generic_visit(node)
```

Comment:
Post-order traversal _just_ fixes this issue, but is mostly a work-around. `_range_is_promotable` should really be changed here. But that's difficult without having to do larger changes.

Reply:
Sounds like that's the right approach then. I'm not sure it's a workaround to be honest.